app/components/publications/batch-documents-publication-modal.js

```diff
@@ -0,0 +1,13 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class PublicationsBatchDocumentsPublicationModalComponent extends Component {
+  get pieces() {
+    return this.args.pieces;
+  }
+
+  @action
+  onCancel() {
+    this.args.onCancel();
+  }
```

Comment:
Waarom deze extra definitie en niet gewoon `@onCancel` gebruiken in de template?

Reply:
Ik vind het qua flow gemakkelijker te begrijpen.
arguments => class <=> template
arguments => class + arguments => template + template <=> class
Maar misschien roept dit voor anderen net meer vragen op. Ik zal het aanpassen.

Reply:
in eerste instantie lijkt dit idd niet nodig.
Ik ben nog niet helemaal mee of we in dit component iets gaan doen dat we moeten annuleren bij sluiten van de modal, dan zou die lokale onCancel een rollback ofzo kunnen doen.

Reply:
@mdv-professional Mijn redenering is als er niets extra moet gebeuren dat het gewoon in de template kan. Dan moet de developer geen extra omweg doen langs de class om te begrijpen wat er gebeurt.

@ValenberghsSven Initieel design was met een opslaan/annuleren knop in de modal. Daarna is het aangepast dat alle acties onmiddelijk effect hebben en er geen opslaan/annuleren meer is. Cancel gaat dus nooit meer moeten doen dan modal sluiten in het huidige design.